request17.go

```diff
@@ -55,6 +55,9 @@ type Request struct {
 // Context method returns the Context if its already set in request
 // otherwise it creates new one using `context.Background()`.
 func (r *Request) Context() context.Context {
+	if r.RawRequest != nil {
```

Comment:
@clippit I have intentionally did not make these changes. Because by resty design PreRequestHook is only meant to modify `RawRequest`  not resty Request` (refer to #62). Any changes made in resty Request instance will not affect the request.